parquet-hadoop/src/main/java/org/apache/parquet/crypto/keytools/KeyToolkit.java

```diff
@@ -206,9 +206,10 @@ String getEncodedWrappedKEK() {
   public static void rotateMasterKeys(String folderPath, Configuration hadoopConfig)
     throws IOException, ParquetCryptoRuntimeException, KeyAccessDeniedException, UnsupportedOperationException {
 
-    if (hadoopConfig.getBoolean(KEY_MATERIAL_INTERNAL_PROPERTY_NAME, KEY_MATERIAL_INTERNAL_DEFAULT)) {
+    if (hadoopConfig.getBoolean(KEY_MATERIAL_INTERNAL_PROPERTY_NAME, false)) {
       throw new UnsupportedOperationException("Key rotation is not supported for internal key material");
     }
+    hadoopConfig.setBoolean(KEY_MATERIAL_INTERNAL_PROPERTY_NAME, false);
```

Comment:
It is not common to set a value in the hadoop conf in our code base. I would expect some comments about why this is necessary. Also, I am not sure if this would not impact the caller. If the caller gives us a hadoop Configuration object we should not modify it.